stan/math/prim/mat/fun/divide_columns.hpp

```diff
@@ -49,17 +24,19 @@ inline typename std::vector<Eigen::Matrix<T_x, R, C>> divide_columns(
  *
  */
 template <typename T_x, typename T_v, int R, int C>
-inline typename std::vector<Eigen::Matrix<T_x, R, C>> divide_columns(
-    const std::vector<Eigen::Matrix<T_x, R, C>> &x,
-    const std::vector<T_v> &vec) {
+inline typename std::vector<
+    Eigen::Matrix<typename return_type<T_x, T_v, double>::type, R, C>>
+divide_columns(const std::vector<Eigen::Matrix<T_x, R, C>> &x,
+               const std::vector<T_v> &vec) {
   size_t N = x.size();
   size_t D = x[0].size();
   check_size_match("divide_columns", "x dimension", D, "vector", vec.size());
 
-  std::vector<Eigen::Matrix<T_x, R, C>> out(N);
+  std::vector<Eigen::Matrix<typename return_type<T_x, T_v, double>::type, R, C>>
+      out(N);
   for (size_t n = 0; n < N; ++n) {
+    out[n].resize(D);
     for (size_t d = 0; d < D; ++d) {
-      out[n].resize(D);
       out[n][d] = divide(x[n][d], vec[d]);
```

Comment:
Is this going to behave correctly if the input is actually matrices? If this is only designed to work with column vectors, limit the signature to working with column vectors.